sympy/functions/special/delta_functions.py

```diff
@@ -191,7 +191,7 @@ def eval(cls, arg, k=0):
         arg = sympify(arg)
         if arg is S.NaN:
             return S.NaN
-        if arg.is_positive or arg.is_negative:
+        if arg.is_positive or arg.is_negative or arg.is_nonzero:
```

Comment:
Instead of this you can do: 
```py
if arg.is_nonzero:
```
This would test all the cases both positive as well as negative.